src/evaluator/matchers/dependency.ts

```diff
@@ -1,27 +1,28 @@
 import { IDependencyMatcherData, MaybeThenable } from '../../dtos/types';
-import { logFactory } from '../../logger/sdkLogger';
 import { IStorageAsync, IStorageSync } from '../../storages/types';
-const log = logFactory('splitio-engine:matcher');
+// import { logFactory } from '../../logger/sdkLogger';
+// const log = logFactory('splitio-engine:matcher');
+import { ILogger } from '../../logger/types';
 import thenable from '../../utils/promise/thenable';
 import { IDependencyMatcherValue, IEvaluation, ISplitEvaluator } from '../types';
 
-function checkTreatment(evaluation: IEvaluation, acceptableTreatments: string[], parentName: string) {
-  let matches = false;
+export default function dependencyMatcherContext(log: ILogger, { split, treatments }: IDependencyMatcherData, storage: IStorageSync | IStorageAsync) {
 
-  if (Array.isArray(acceptableTreatments)) {
-    matches = acceptableTreatments.indexOf(evaluation.treatment as string) !== -1;
-  }
+  function checkTreatment(evaluation: IEvaluation, acceptableTreatments: string[], parentName: string) {
+    let matches = false;
 
-  log.debug(`[dependencyMatcher] Parent split "${parentName}" evaluated to "${evaluation.treatment}" with label "${evaluation.label}". ${parentName} evaluated treatment is part of [${acceptableTreatments}] ? ${matches}.`);
+    if (Array.isArray(acceptableTreatments)) {
+      matches = acceptableTreatments.indexOf(evaluation.treatment as string) !== -1;
+    }
 
-  return matches;
-}
+    log.debug(`[dependencyMatcher] Parent split "${parentName}" evaluated to "${evaluation.treatment}" with label "${evaluation.label}". ${parentName} evaluated treatment is part of [${acceptableTreatments}] ? ${matches}.`);
 
-export default function dependencyMatcherContext({ split, treatments }: IDependencyMatcherData, storage: IStorageSync | IStorageAsync) {
+    return matches;
+  }
 
   return function dependencyMatcher({ key, attributes }: IDependencyMatcherValue, splitEvaluator: ISplitEvaluator): MaybeThenable<boolean> {
     log.debug(`[dependencyMatcher] will evaluate parent split: "${split}" with key: ${JSON.stringify(key)} ${attributes ? `\n attributes: ${JSON.stringify(attributes)}` : ''}`);
-    const evaluation = splitEvaluator(key, split, attributes, storage);
+    const evaluation = splitEvaluator(log, key, split, attributes, storage);
```

Comment:
This is the one we need to double check we're solving for, if we remove the logs from the matchers.